src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java

```diff
@@ -141,4 +149,34 @@ public ILobby getLobby(String token, Long lobbyId){
         throw new ResponseStatusException(HttpStatus.FORBIDDEN, errorMessage);
     }
 
+    public void updateLobby(ILobby lobby, String token, String lobbyName, Visibility visibility, GameMode gameMode, GameType gameType) {
+        if (!token.equals(lobby.getHost().getToken())) {
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "User is not the host of the lobby.");
```

Comment:
I would distinguish between (token == null || token.empty()) and (!token.equals(lobby.getHost().getToken()).
Also UNAUTHORIZED means "unauthenticated" eg. if the token is null or empty. FORBIDDEN is if the token is set but does not match.